test/lib/translators/ui5Framework.js

```diff
@@ -0,0 +1,228 @@
+			{fake: "metadata-project-3"}
+		]
+	});
+});
+
+test.serial("generateDependencyTree should ignore root project without framework configuration", async (t) => {
+	const tree = {
+		id: "test-id",
+		version: "1.2.3",
+		path: "/test-project/",
+		metadata: {
+			name: "test-name"
+		},
+		dependencies: []
+	};
+	const ui5FrameworkTree = await ui5Framework.generateDependencyTree(tree);
+
+	t.is(ui5FrameworkTree, null, "No framework tree should be returned");
+});
+test.serial("utils.isFrameworkProject", (t) => {
+	t.true(utils.isFrameworkProject({id: "@sapui5/foo"}), "@sapui5/foo");
+	t.true(utils.isFrameworkProject({id: "@openui5/foo"}), "@openui5/foo");
+	t.false(utils.isFrameworkProject({id: "sapui5"}), "sapui5");
+	t.false(utils.isFrameworkProject({id: "openui5"}), "openui5");
+});
+test.serial("utils.shouldIncludeDependency", (t) => {
+	// root project dependency should always be included
+	t.true(utils.shouldIncludeDependency({}, true));
+	t.true(utils.shouldIncludeDependency({optional: true}, true));
+	t.true(utils.shouldIncludeDependency({optional: false}, true));
+	t.true(utils.shouldIncludeDependency({optional: null}, true));
+	t.true(utils.shouldIncludeDependency({optional: "abc"}, true));
+	t.true(utils.shouldIncludeDependency({development: true}, true));
+	t.true(utils.shouldIncludeDependency({development: false}, true));
+	t.true(utils.shouldIncludeDependency({development: null}, true));
+	t.true(utils.shouldIncludeDependency({development: "abc"}, true));
+	t.true(utils.shouldIncludeDependency({foo: true}, true));
+
+	t.true(utils.shouldIncludeDependency({}, false));
+	t.false(utils.shouldIncludeDependency({optional: true}, false));
+	t.true(utils.shouldIncludeDependency({optional: false}, false));
+	t.true(utils.shouldIncludeDependency({optional: null}, false));
+	t.true(utils.shouldIncludeDependency({optional: "abc"}, false));
+	t.false(utils.shouldIncludeDependency({development: true}, false));
+	t.true(utils.shouldIncludeDependency({development: false}, false));
+	t.true(utils.shouldIncludeDependency({development: null}, false));
+	t.true(utils.shouldIncludeDependency({development: "abc"}, false));
+	t.true(utils.shouldIncludeDependency({foo: true}, false));
+
+	// Having both optional and development should not be the case, but that should be validated beforehand
+	t.true(utils.shouldIncludeDependency({optional: true, development: true}, true));
+	t.false(utils.shouldIncludeDependency({optional: true, development: true}, false));
+});
+test.serial("utils.getFrameworkLibrariesFromTree: Project without dependencies", (t) => {
+	const tree = {
+		id: "test",
+		metadata: {
+			name: "test"
+		},
+		framework: {
+			libraries: []
+		},
+		dependencies: []
+	};
+	const ui5Dependencies = utils.getFrameworkLibrariesFromTree(tree);
+	t.deepEqual(ui5Dependencies, []);
+});
+
+test.serial("utils.getFrameworkLibrariesFromTree: Project with libraries and dependency with libraries", (t) => {
+	const tree = {
```

Comment:
Either add some "UI5 Framework libraries" (like `@sapui5/zebra`) or a stub for `utils.isFrameworkProject` to check whether `getFrameworkLibrariesFromTree` correctly calls `utils.isFrameworkProject`